### DISCUSSION TITLE
rootDepth versus searched depth.

### DISCUSSION THREAD

**Discussion by vondele** [official-stockfish/Stockfish#3888] [Show and tell]
The question how 'deep' the engine searches for a given 'depth' specified at the root is a recurring one. 

The following graph shows how many nodes are visited in search at a given distance from the root (ply from root) for a given rootDepth. 

![ply_from_root](https://user-images.githubusercontent.com/4202567/149080827-fb585b88-3a69-41b9-8dfb-634c3a71bc92.png)

The data has been collected running a depth 22 search over a collection of fens.

As expected, the average 'ply from root' increases with rootDepth, peaks somewhat before the nominal value, and extends to much larger values.

**Comment by Dboingue** [comment]
Enfin!  I had been craving for such picture for a while.  Thanks for sharing and publishing.  The  SF entrails statistics can help making sense from the user point of view (and it might help  the dev. as well, or the new collabs, hypothesis.).   This is what I mean by bringing the documentation to chess land.  A piece of it at least.

**Comment by Dboingue** [comment]
Ok. I like it so much that I must ask to see the raw data. or some link to it.. I wish to get a sense of the variability as a function of FEN input.
That would be too  much to put on this graph, I understand.  But it would help get a sense of the bounds on the left vertical axis.

I was also wondering, if the same data, was also containing the nodes that finally did get a leaf evaluation among all the nodes visited. If one could do statistics on that compared to other nodes.  

I just realized this was a discussion thread..    I hope not too late.  Good direction, I insist.

Rephrasing possibly better my last question about leaf evaluations..  First could you drop a line about what "visited" node definition?
Then from there, the leaf evaluation node type question could make more sense.

I  am hesitating about the meaning and extent of information that one could interpret from the graph about some model of tree searches in chess land (not code, but still a graphical tree model in mind, as I think this graph is also implying, which I am liking).

I know or think I understand from previous battles with existing chess engine literature or documentation**, that there are many type of nodes being recursively considered and dispatched to various heuristics before such nodes make it to the leaf evaluation base conditions.

And various indices being grown also, including extension searches selecting for something called quiescence criterion (which I would approximate as a position without any tension remaining-- need check or confirmation of understanding here, please).

I am not asking for a full deployment of all node types.. I am currently assuming that visited includes all types.  Hence the ask about discerning those that are node having had leaf evaluation (simple, or NNue all of them combined if that is all we can get).
First chess land question of interest to me, is looking at a done tree search (assuming you had to do that anyway, keep that in spite of its speed cost) which nodes did not get an eval, and which did. As a summary of the selectivity of the tree search for leaf evaluations.  

Motivation behind being that in all optimization process over all past SF versions only such positions would have an optimization contribution to all evalation parameters under fishtest and tournament ELO measurements.

** (not really including SF most recent laudable wiki efforts), but I think discussion is more efficient for me and for others as well, bringing up the more critical notions needed for communication across different background or training assumptions, we all have different versions of what is obvious for example). So my questions are trying to help spilling those obvious beans, if I can talk that way without offense.

I could not resist explaining why..  sorry.  This might be another discussion.. my motivation itself might be foggy.. but I hope the node type question and request for the source data whereabouts. is not..
